helix-term/src/ui/mod.rs

```diff
@@ -78,25 +78,44 @@ pub fn regex_prompt(
 
 pub fn file_picker(root: PathBuf) -> Picker<PathBuf> {
     use ignore::Walk;
+    use std::time;
     let files = Walk::new(root.clone()).filter_map(|entry| match entry {
         Ok(entry) => {
             // filter dirs, but we might need special handling for symlinks!
             if !entry.file_type().map_or(false, |entry| entry.is_dir()) {
-                Some(entry.into_path())
+                let time = if let Ok(metadata) = entry.metadata() {
+                    if let Ok(atime) = metadata.accessed() {
+                        atime
+                    } else if let Ok(mtime) = metadata.modified() {
+                        mtime
+                    } else if let Ok(ctime) = metadata.created() {
+                        ctime
+                    } else {
+                        time::UNIX_EPOCH
+                    }
+                } else {
+                    time::UNIX_EPOCH
+                };
+
+                Some((entry.into_path(), time))
             } else {
                 None
             }
         }
         Err(_err) => None,
     });
 
-    let files = if root.join(".git").is_dir() {
+    let mut files: Vec<_> = if root.join(".git").is_dir() {
         files.collect()
     } else {
         const MAX: usize = 8192;
         files.take(MAX).collect()
     };
 
+    files.sort_by(|(_, time1), (_, time2)| time1.cmp(time2));
+
+    let files = files.into_iter().map(|(path, _)| path).collect();
```

Comment:
Can we collect into two vec using `unzip`. Then, `zip` it into an iterator and use `sort_by_key` from the second iterator. So with this method we can remove the `map` and `collect` to files. What do you think?

Both methods collect twice, but the one I suggested collects beforehand.

Reply:
Can you sort an iterator though? I was sure that the methods are not available.

Reply:
Also, as a side note, I think this approach is a bit cleaner, but I'll let others chime in to decide.

Reply:
Wait no, sorry. Don't `zip` the iterator but just sort using the slice. Both approach seemed similar to me with the one I suggested taking less lines of code.

Reply:
Yes, now we can use `sort_by_key`. Done.